### PR TITLE
Update tests

### PR DESCRIPTION
**Pull request by hfrick**
Testing the model type (`proportional_hazards()` and `survival_reg()`) is better done in parsnip than censored. I looked at moving the tests to parsnip but: parsnip used to have them and got updated so I don't think it makes sense to re-add them.

The heritage tests on model fitting / execution for `survival_reg()` (inheriting from `surv_reg()`) have been updated to check the resulting model fit object, like the tests for the other model type / engines in censored.

### CODE REVIEW

**Comment by hfrick** [comment]
After our discussion I've added tests for primary args and engine args back in for the two model types which do not have any engines in parsnip: `survival_reg()` and `proportional_hazards()`, see https://github.com/tidymodels/parsnip/pull/791

File: tests/testthat/test-survival_reg-flexsurv.R [hunk 1]
```diff
@@ -1,25 +1,28 @@
 library(testthat)
 
-test_that("flexsurv execution", {
-  expect_error(
-    res <- survival_reg(dist = "lognormal") %>%
```
Comment (simonpcouch) [inline tests/testthat/test-survival_reg-flexsurv.R:5]: A question more for my own understanding than about correctness:

There are four removals of this `dist = "lognormal"` argument but no additions including it. Are we still testing this somewhere / do we need to? Is this a default now?

**Review by simonpcouch — APPROVED** [review]
This looks super clean! Hope you feel that the helpers are a helpful tool. One comment, otherwise thumbs up from me. :)

File: tests/testthat/test-survival_reg-flexsurv.R (hunk 1 above)
Comment (hfrick) [reply tests/testthat/test-survival_reg-flexsurv.R:5]: yes, a different test (the one on survival time predictions) still uses a non-default distribution 👍

**Pull request merged**